bite-size-notes/week-9-a-Extended-Kalman-Filter.md

```diff
@@ -0,0 +1,130 @@
+<h1>Extended Kalman Filter</h1>
+
+<h3>Objectives</h3>
+
+The objectives of these notes is to highlight the important differences when applying an extended kalman filter as compared to the standard linear kalman filter.
+
+<h3>Introduction</h3>
+
+The Extended Kalman Filter (EKF) is a powerful extension of the Kalman Filter designed to handle nonlinear systems. While the standard Kalman Filter excels in linear scenarios, many real-world systems exhibit nonlinear behavior that the standard filter cannot address adequately. The EKF overcomes this limitation by linearizing the nonlinear system around the current estimate using a first-order Taylor series expansion at each time step.
+
+<h3>Preliminaries: Notation and problem setup</h3>
+
+We will assume Nonlinear Discrete Dynamics with zero mean gaussian process and measurement noise
+
+$$\begin{align*}
+x_{k+1}&=f(x_k,u_k,w_k) &&\qquad w_k\sim N(0,Q)\\
+y_{k+1}&=g(x_k,v_k) &&\qquad v_k\sim N(0,R)
+\end{align*}$$
+
+where f() describes how the dynamics propagate forward in time and g() describes how the states map to measurements of the system.
+
+At each step we will linearize and use the standard kalman filter equations. To linearize we will take the jacobian of both f() and g() w.r.t. the state x, and noise w/v.
+
+$F_{k}^{x}=\frac{\partial f}{\partial x}|_{x_k,u_k,w_k}$
+$F_{k}^{w}=\frac{\partial f}{\partial w}|_{x_k,u_k,w_k}$
+$G_{k}^{x}=\frac{\partial g}{\partial x}|_{x_k,u_k,v_k}$
+$G_{k}^{v}=\frac{\partial g}{\partial v}|_{x_k,u_k,v_k}$
+
+Here subscript denotes the timestep and the superscript denotes what the jacobian is being taken with respect to. Additionally if T is in the superscript like $G_{k}^{xT}$, then that means it is the transpose of $G_{k}^{x}$
+
+For generalization it is assumed that the noise is not additive. If the noise is additive then the partials with respect to w or v will be identity matrices.
+
+<h3>General Extended Kalman Filter Algorithm</h3>
+
+Predict:
+
+$$
+\begin{align*}
+
+\mu_{k+1}^p&=f(\mu_k,u_k)\\
+
+\Sigma_{k+1}^p&=F_k^x\Sigma_k F_k^{xT}+F_k^wQF_k^{wT}\\
+
+y_{k+1}^p&=g(\mu_{k+1}^p)
+\end{align*}
+$$
+
+Update:
+
+$$\begin{align*}
+
+K_{k+1}&=\Sigma_{k+1}^pG_{k+1}^{xT}(G_{k+1}^{x}\Sigma_{k+1}^p G_{k+1}^{xT}+G_{k+1}^{v}RG_{k+1}^{vT})^{-1}\\
+
+\mu_{k+1}&=\mu_{k+1}^p+K_{k+1}(y_{k+1}-y_{k+1}^p)\\
+
+\Sigma_{k+1}&=\Sigma_{k+1}^p-K_{k+1}(G_{k+1}^x\Sigma_{k+1}^pG_{k+1}^{xT}+G_{k+1}^vRG_{k+1}^{vT})K_{k+1}^T
+
+\end{align*}$$
+
+Notice that we use the full nonlinear system in the mean prediction step but we have to use the linearized system when looking at how the 

```

Comment:
That is a valid point but ends abruptly. Need to complete and explain the difference.

Reply:
Thanks, I don't think the latest copy had been saved when published. Fixed.